halo2_proofs/src/dev/cost_model.rs

```diff
@@ -353,16 +353,25 @@ pub fn from_circuit_to_cost_model_options<F: Ord + Field + FromUniformBytes<64>,
         let mut expanded_rows_count = 0;
         let mut compressed_rows_count = 0;
         for region in prover.regions {
-            if region.name.contains("table") { // Account for lookup tables separately
-                let (start, end) = region.rows.expect("region.rows should be known by now");
-                min_k = std::cmp::max(min_k, end - start);
-            } else {
-                let (start, end) = region.rows.expect("region.rows should be known by now");
-                expanded_rows_count = expanded_rows_count + ((end - start) + 1);
-                compressed_rows_count = std::cmp::max(compressed_rows_count, end + 1);
-            }
+            // If `region.rows == None`, then that region has no rows (e.g. just copy constraints)
+            if let Some((start, end)) = region.rows {
+                // A region is a _table region_ if all of its columns are `Fixed`
+                // columns (see that [`plonk::circuit::TableColumn` is a wrapper
+                // around `Column<Fixed>`]). All of a table region's rows are
+                // counted towards `table_rows_count.`
+                if region
+                    .columns
+                    .iter()
+                    .all(|c| *c.column_type() == crate::plonk::Any::Fixed)
+                {
+                    min_k = std::cmp::max(min_k, end - start);
+                } else {
+                    expanded_rows_count += (end - start) + 1;
```

Comment:
I know this was old code, but why the + 1?

Reply:
Fixed in [970bb6d](https://github.com/input-output-hk/halo2/pull/3/commits/970bb6d72f9dc0e1661e1c9450cbb4dda7448278), also added explanation of why.